### PR TITLE
Update KafkaMessageListenerContainer.java

### PR DESCRIPTION
**Pull request by yangqiju**
destroy thread pool.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@yangqiju Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=491)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=491) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@yangqiju Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=491)!

File: spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java [hunk 1]
```diff
@@ -663,6 +663,9 @@ public void run() {
 				KafkaMessageListenerContainer.this.stop();
 			}
 			this.monitorTask.cancel(true);
+			if (this.taskScheduler instanceof ThreadPoolTaskScheduler) {
+				((ThreadPoolTaskScheduler) this.taskScheduler).destroy();
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java]: I see why you do this:
```
if (this.containerProperties.getScheduler() != null) {
	this.taskScheduler = this.containerProperties.getScheduler();
}
else {
	ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
	threadPoolTaskScheduler.initialize();
	this.taskScheduler = threadPoolTaskScheduler;
}
```
So, we create an internal non-managed instance if it isn't provided via `ContrainerProperties`.

Well, that's indeed a good catch.
And thank you very much for the contribution on the matter!

Only the problem that we can't do that unconditionally.

In other places we have some `boolean` flag to control injection/internal instance. Something like `taskSchedulerExplicitlySet`. And call this `destroy()` only if it is `false`.

**Comment by yangqiju** [comment]
Thank you very much.

**Comment by artembilan** [comment]
Why did you close it? 
I told this is a good fix, but just needs some improvements.
You can just push additional commits to your `patch-1` branch.

**Comment by yangqiju** [comment]
Oh, I think I understand what you mean.
I'll try to improve later.
thanks.

**Comment by artembilan** [comment]
Merged and cherry-picked as https://github.com/spring-projects/spring-kafka/commit/8c65d62a247a0f247ccade8c8a060113bc010937

Thank you for contribution.
Looking forward for more!

**Pull request closed without merging**